### DISCUSSION TITLE
Connect the wavesurfer with other players like vidstack

### DISCUSSION THREAD

**Discussion by woss** [katspaugh/wavesurfer.js#3980] [Q&A]
I'm using vidstack for all my multimedia playbacks and it would be great to have a connection between the two 
For example, I could play the music from the vidstack player and the wavesurfer would somehow know ( through the event )  on which position the song is.

I'm not asking that someone does this for me, I am looking for the pointers how can I do it myself :) 

<img width="415" alt="image" src="https://github.com/user-attachments/assets/2e96365f-1dc1-4daa-bce2-f0391c30180e" />

**Comment by katspaugh** [comment]
You can make the two players share the same audio element.

Get the underlying audio element from vidstack:
https://vidstack.io/docs/player/api/providers/audio/?styling=default-theme#element

Then pass it to wavesurfer in the `media` property as shown here:
https://wavesurfer.xyz/examples/?video.js
